Tools/WinMLRunner/src/BindingUtilities.cpp

```diff
@@ -646,6 +646,28 @@ namespace BindingUtilities
         }
     }
 
+    TensorString CreateTensorString(const std::vector<int64_t>& tensorShape)
+    {
+        auto length = 1;
+        for (auto&& dim : tensorShape)
+        {
+            if (dim > 0)
```

Comment:
Is this the same behavior as the other types of tensors? It feels like we should either fail early or keep the 0's in the multiplications (empty tensors are valid in ORT, although it wouldn't be very useful for an input). Ignoring the 0 would give us a "valid" shape, but probably not what the caller intended.

Reply:
Are there use cases for 0 size tensors? I don't think I've seen those. This is mostly to catch when a model has "-1" to symbolize a free dimension, but multiplying by -1 would obviously cause problems in the size calculation. In those cases, we just pick "1" as the free dimension since it's impossible to know what the actual use case is in a generic test tool.

Reply:
Already submitted, but I want to dispel the myth that these two things are invalid:

- 0 size tensors are valid - they're simply empty tensors. In most cases, operator execution on them is a nop (just like `memset(p, 0, 0)` being a nop), but they can be the output of a slice that sliced away everything, and they can be the input to a padding operation (which then yields a non-empty tensor).
- 0D tensors are valid - they're just scalar values.